src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -219,15 +354,423 @@ private function isDeclareStatement(Tokens $tokens, $index)
+                            ? $tokens->getNextMeaningfulToken($i + 1)
+                            : $i + 1
+                        ;
+                        $blockType = Tokens::detectBlockType($tokens[$arrayStartIndex]);
+                        $arrayEndIndex = $tokens->findBlockEnd($blockType['type'], $arrayStartIndex);
+
+                        if ($tokens->isPartialCodeMultiline($arrayStartIndex, $arrayEndIndex)) {
+                            break;
+                        }
+                    }
+
+                    ++$index;
+                }
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $from
+     * @param int    $until
+     */
+    private function injectArrayAlignmentPlaceholders(Tokens $tokens, $from, $until)
+    {
+        // Only inject placeholders for multi-line arrays
+        if ($tokens->isPartialCodeMultiline($from, $until)) {
+            ++$this->deepestLevel;
+            ++$this->currentLevel;
+            $this->injectAlignmentPlaceholdersForArrow($tokens, $from, $until);
+            --$this->currentLevel;
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     * @param string $alignStrategy
+     */
+    private function fixWhiteSpaceBeforeOperator(Tokens $tokens, $index, $alignStrategy)
+    {
+        // fix white space after operator is not needed as BinaryOperatorSpacesFixer took care of this (if strategy is _not_ ALIGN)
+        if ($tokens[$index - 1]->isWhitespace()) {
+            if (self::ALIGN_SINGLE_SPACE_MINIMAL === $alignStrategy) {
+                $content = $tokens[$index - 1]->getContent();
+                if (' ' !== $content && false === strpos($content, "\n") && !$tokens[$tokens->getPrevNonWhitespace($index - 1)]->isComment()) {
+                    $tokens[$index - 1] = new Token([T_WHITESPACE, ' ']);
+                }
+            }
+        } else {
+            $tokens->insertAt($index, new Token([T_WHITESPACE, ' ']));
+        }
+    }
+
+    /**
+     * Look for group of placeholders and provide vertical alignment.
+     *
+     * @param Tokens $tokens
+     * @param string $alignStrategy
+     *
+     * @return string
+     */
+    private function replacePlaceholders(Tokens $tokens, $alignStrategy)
     {
-        /** @var AbstractAlignFixerHelper $helper */
-        foreach ($this->alignFixerHelpers as $helper) {
-            if ($tokens->isChanged()) {
-                $tokens->clearEmptyTokens();
+        $tmpCode = $tokens->generateCode();
+
+        for ($j = 0; $j <= $this->deepestLevel; ++$j) {
+            $placeholder = sprintf(self::ALIGN_PLACEHOLDER, $j);
+
+            if (false === strpos($tmpCode, $placeholder)) {
+                continue;
             }
 
-            $helper->fix($tokens);
+            $lines = explode("\n", $tmpCode);
+            $groupIndex = [];
+            $blockSize = 0;
```

Comment:
I think you renamed the wrong variable after https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2573#discussion_r134124989.

Reply:
if $blockSize -> $groupIndex, then curren $groupIndex -> ?

Reply:
Wasn't current `$groupIndex` named `$groups` before?

Reply:
:+1: done